src/com.frostycoolslug.goxlr-utility.sdPlugin/app/toggle-mute.js

```diff
@@ -12,12 +12,16 @@ function muteToggleExternalStateChange() {
 
 muteToggle.onKeyDown(({action, event, context, device, payload}) => {
     let longPressDelay = 500; // default value
+    let faderMuteStatus = null;
     if (status !== undefined && status.mixers[payload.settings.serial] !== undefined) {
         longPressDelay = status.mixers[payload.settings.serial].settings.mute_hold_duration;
+        faderMuteStatus = status.mixers[payload.settings.serial].fader_status[payload.settings.fader].mute_state;
     }
 
     longPress = setTimeout(() => {
-        doMute(context, payload.settings.serial, payload.settings.fader, payload.settings.mode, "MutedToAll");
+        if (faderMuteStatus === "Unmuted") {
```

Comment:
I only have a couple of minutes, but I'm going to need to review this closer..

From what I can tell, if a user has the `payload.settings.mode` configured as 'set' with 'MuteToAll' as the target, if the channel is currently MutedToX and they hold the button it'll do nothing at all? It should go ToAll as soon as they release the button instead.

It might be easer to have checks in the `onKeyDown` and `onKeyUp` to confirm the mode before handling the behaviour..

so `onKeyDown` if `mode === 'set'` just do nothing, and `onKeyUp` if the mode is set, force send a `doMute`, then include your code for holding into the other branches so it only works when the mode is toggle.

Reply:
I just tested, the behaviour on the GoXLR is
1) press -> MuteToX
2) press&Hold -> MuteToAll
3) press&hold -> nothing
4) press -> Unmute

Reply:
Apologies, I think I misunderstood the nature of this commit, looks like it'll only handle the hold state if the channel is unmuted, rather than resending it (which is reasonable)..

The comment is still appropriate for `mode === 'set'`, just needs to make sure that doMute is always sent in that scenario (doMute itself should handle the state)..

I'm kinda swamped at the moment, so I'll try and get to this as I can, just a little rushed :D

Reply:
no problem, i'm in no rush. you get to it when you can. I'll take another look at it myself

Reply:
I just looked into it, you are correct. I didn't realize there was a "set" mode, and it is totally broken with my change. I'll fix and update the PR

Reply:
The behaviour on the GoXLR also depends on the initial state..

Unmuted -> Press -> MuteToX
Umuted -> Hold -> MuteToAll
MutedToX -> Press -> Unmuted
MutedToX -> Hold -> MuteToAll
MutedToAll -> Press -> Unmuted
MutedToAll -> Hold -> Do nothing

It might make more sense to have these behaviours defined in doMute function (around line 77), and have that verify and update accordingly based on what's passed in, and have 'toggle' mode be more 'replicate the GoXLR'..

Just spent some time double checking the 'set' behaviour, and it'll override the value passed in with that set in config, but it should be called in *ALL* situations so that regardless of the current state, or how it got there. I think removing this if and moving the logic into doMute would solve both these.

Reply:
yeah, I'll re-write as such

Reply:
I've run into an interesting problem... if it is MuteToAll and you try to set->MuteToX, you get an error, you can't transition MuteToAll->MuteToX

Reply:
Correct, it was a bit of an oversight that happened in the early days of the utility that's actually pretty painful to fix now. When the utility was designed it was focusing on button behaviour, and so didn't get extra code for that transition (because it's not possible to do that on the physical GoXLR using buttons), so when it came to implementing that via the API it wasn't simple.

The easiest way to deal with it for now is to transition via unmute